2017/opmodes/VioletJewelAutonomous.java

```diff
@@ -0,0 +1,363 @@
+                    togglePolling();
+                    break;
+                // case RIGHT_TRIGGER_DOWN:
+                //   toggleFlash();
+                //    break;
+                default:
+                    break;
+            }
+        }
+    }
+
+    private void togglePolling() {
+        if (pollOn == false) {
+            colorThiefTask.setPollingMode(ColorThiefTask.PollingMode.ON);
+            pollOn = true;
+
+        } else {
+            colorThiefTask.setPollingMode(ColorThiefTask.PollingMode.OFF);
+            pollOn = false;
+        }
+    }
+
+   /*  Currently not using flash! Front camera doesn't have one.
+
+   private void toggleFlash() {
+        if (flashOn == false) {
+            CameraDevice.getInstance().setFlashTorchMode(flashOn);
+            flashOn = true;
+            flashItem.setValue("On");
+        } else {
+            CameraDevice.getInstance().setFlashTorchMode(!flashOn);
+            flashOn = false;
+            flashItem.setValue("Off");
+        }
+    }
+    */
+
+    private void sense()
+    {
+         colorThiefTask = new ColorThiefTask(this) {
+            @Override
+            public void handleEvent(RobotEvent e) {
+                ColorThiefTask.ColorThiefEvent event = (ColorThiefEvent) e;
+                particle.setValue(event.toString());
+
+                pushJewel = new DeadReckonPath();
+
+                if (alliance == Alliance.RED) {
+                    if (event.kind == EventKind.RED) {
+                        pushJewel.stop();
+                        RobotLog.i("506 Sensed RED");
+                        pushJewel.addSegment(DeadReckonPath.SegmentType.TURN, 35, Violet.TURN_SPEED * turnMultiplier);
+                        RobotLog.i("506 First turn done");
+                        pushJewel.addSegment(DeadReckonPath.SegmentType.TURN, 35, Violet.TURN_SPEED);
+                        RobotLog.i("506 Second turn done");
+                    } else {
+                        RobotLog.i("506 Sensed BLUE");
+                        //pushJewel.stop();
+                        //pushJewel.addSegment(DeadReckonPath.SegmentType.STRAIGHT, 5, Violet.STRAIGHT_SPEED * moveMultiplier);
+                    }
+                } else {
+                    if (event.kind == EventKind.BLUE) {
+                        RobotLog.i("506 Sensed BLUE");
+                        pushJewel.stop();
+                        pushJewel.addSegment(DeadReckonPath.SegmentType.TURN, 35, Violet.TURN_SPEED * turnMultiplier);
+                        pushJewel.addSegment(DeadReckonPath.SegmentType.TURN, 35, Violet.TURN_SPEED);
+                    } else {
+                        RobotLog.i("506 Sensed RED");
+                        //pushJewel.stop();
+                        //pushJewel.addSegment(DeadReckonPath.SegmentType.STRAIGHT, 5, -Violet.STRAIGHT_SPEED);
+                    }
+                }
+
+            }
+        };
+
+        addTask(colorThiefTask);
+        addTask(stt);
```

Comment:
Remember this addTask call returns immediately.  So it's not actually doing the wait that you want it to.